File: besoinEssai/src/main/java/com/example/besoinEssai/model/Commentaire.java
package com.example.besoinEssai.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import lombok.Data;

import javax.persistence.*;
import java.util.Calendar;
import java.util.Date;

@Entity
@Data
public class Commentaire {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long commentaireId;
    public String texteCommentaire;
    public Date dateCommentaire;
    private String utilisateurEmetteur;
    @ManyToOne(cascade = CascadeType.PERSIST)
    private Utilisateur utilisateurDestinataire;
    @ManyToOne
    @JsonBackReference
    private Besoin besoin;
    private boolean estLu;
}

File: besoinEssai_front/src/app/models/Commentaire.ts
import { Utilisateur } from "./Utilisateur"
import { Besoin } from "./besoin"

export interface Commentaire {
    commentaireId?: number,
    texteCommentaire?: string,
    dateCommentaire: Date,
    utilisateurEmetteur?: string,
    utilisateurDestinataire?: Utilisateur,
    besoin?: Besoin,
    estLu?:boolean;
}